### ISSUE TITLE
picar.jar

### ISSUE THREAD

**Issue by ricardoi** [MatthewBareno/poplar-septoria-GWAS#2]
Try `RGID= Sample_name RGLB= Sample_name RGPL= Illumina RGPU= Sample_name RGSM=Sample_name`

**Comment by MatthewBareno** [comment]
[Linux@anduin test]$ java -jar ../../../programs/picard.jar AddOrReplaceReadGroups -I Nisk1.sam -O CleanedNisk11.sam -RGID Nisk1 -RGLB Nisk1 -RGPL Illumina -RGPU Nisk1 -RGSM Nisk1
11:42:02.631 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/raid1/home/bpp/barenom/programs/picard.jar!/com/intel/gkl/native/libgkl_compression.so
[Tue Jul 12 11:42:02 PDT 2022] AddOrReplaceReadGroups --INPUT Nisk1.sam --OUTPUT CleanedNisk11.sam --RGID Nisk1 --RGLB Nisk1 --RGPL Illumina --RGPU Nisk1 --RGSM Nisk1 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
[Tue Jul 12 11:42:02 PDT 2022] Executing as barenom@anduin.cgrb.oregonstate.local on Linux 3.10.0-1160.62.1.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_71-b15; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT
INFO	2022-07-12 11:42:02	AddOrReplaceReadGroups	Created read-group ID=Nisk1 PL=Illumina LB=Nisk1 SM=Nisk1

[Tue Jul 12 11:42:02 PDT 2022] picard.sam.AddOrReplaceReadGroups done. Elapsed time: 0.00 minutes.
Runtime.totalMemory()=2058354688
To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp
Exception in thread "main" htsjdk.samtools.SAMFormatException: Error parsing text SAM file. Empty field at position 9 (zero-based); File Nisk1.sam; Line 80
Line: J00107:161:HNCCKBBXX:1:1101:31649:1156	133	NW_016017386.1	612910	0	*	=	612910	0		*	MC:Z:55M	AS:i:0	XS:i:0
	at htsjdk.samtools.SAMLineParser.reportErrorParsingLine(SAMLineParser.java:457)
	at htsjdk.samtools.SAMLineParser.parseLine(SAMLineParser.java:238)
	at htsjdk.samtools.SAMTextReader$RecordIterator.parseLine(SAMTextReader.java:268)
	at htsjdk.samtools.SAMTextReader$RecordIterator.next(SAMTextReader.java:255)
	at htsjdk.samtools.SAMTextReader$RecordIterator.next(SAMTextReader.java:228)
	at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:591)
	at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:570)
	at picard.sam.AddOrReplaceReadGroups.doWork(AddOrReplaceReadGroups.java:183)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:308)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:113)

**Comment by ricardoi** [comment]
After the adding or replacing groups with `picard.jar`, execute the following commands:
```bash
picard.jar SortSam -I ID.sam -O ID.rg.bam -SORT_ORDER coordinate -CREATE_INDEX true
picard.jar MarkDuplicates -I ID.rg.bam -O ID.mdup.bam -ASSUME_SORT_ORDER coordinate -CREATE_INDEX true
picard.jar SortSam -I ID.mdup.bam -O ID.mdup.sorted.bam -SORT_ORDER coordinate -CREATE_INDEX true
```
Then execute the command `gatk HaplotypeCaller`
